cmd/earthly/build_cmd.go

```diff
@@ -374,9 +374,10 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 		imageResolveMode = llb.ResolveModeForcePull
 	}
 
-	cacheImports := make(map[string]bool)
-	if app.remoteCache != "" {
-		cacheImports[app.remoteCache] = true
+	cacheImports := make([]string, 0)
+	if app.remoteCache != "" || len(app.cacheFrom.Value()) > 0 {
```

Comment:
this seems wrong. I would do these as two seperate IFs:

```
if app.remoteCache != "" {
   cacheImports = append(cacheImports, app.remoteCache)
}
if len(app.cacheFrom.Value()) > 0 {
  cacheImports = append(cacheImports, app.cacheFrom.Value()...)
}
```

Reply:
Fix by [cce0427](https://github.com/earthly/earthly/pull/2063/commits/cce0427591aca7aee458fb88b6d642d2b7d3df3e)